ggml/src/ggml-cuda/ggml-cuda.cu

```diff
@@ -3501,13 +3556,35 @@ static void evaluate_and_capture_cuda_graph(ggml_backend_cuda_context * cuda_ctx
                 }
 #else
                 GGML_UNUSED(integrated);
-#endif // NDEBUG
+#endif  // NDEBUG
 
                 bool ok = ggml_cuda_compute_forward(*cuda_ctx, node);
                 if (!ok) {
                     GGML_LOG_ERROR("%s: op not supported %s (%s)\n", __func__, node->name, ggml_op_name(node->op));
                 }
                 GGML_ASSERT(ok);
+
+                if (!is_concurrent_event_active) {
+                    //const ggml_tensor * adjusted_node = node;
+                    // the forking node may have been fused, e.g (RMS_NORM_MUL + MUL + ADD),
+                    // we can safely use the previous node to check if it can be forked
```

Comment:
```suggestion
```

I feel this comment belongs to lines 3233 (where we compare and search for previous node). This code here triggers only when `GGML_CUDA_DISABLE_FUSION=1`